### DISCUSSION TITLE
"zellij run --" how to run command with aliases

### DISCUSSION THREAD

**Discussion by waldauf** [zellij-org/zellij#2701] [Q&A] [answered]
Hi,

I would like to run open a new pane in zellij from the command line: `zellij run -f -- <command with alias>`. But I got stuck on alias evaluation:
```
zellij run -f -- gss

Command not found: gss
If you were including arguments as part of the command, try including them as 'args' instead. 
```
Where `gss='git status --short'` 

Is there a way how to open new shell with all variables/aliases?

**Comment by Lockszmith-GH** [comment]
This is a shell issue, not a zellij one. You didn't mention which shell you are using, so I'll asume it's bash or zsh, which behave the same here.

`alias` is a shell construct, `zellij run` can only execute executables from `$PATH`.

The way around this is to have the shell exapnd the alias before actually executing `zellij`.

By default, the shell will expand other aliases on the command line before exectuion of an alias. So let's make `zellij` and alias first:
```
alias zellij='zellij '
alias gss='git status --short '
```
then the command you where trying to run will expand and execute as you wanted.

BTW: This is what a lot of people do with sudo as well:
```
alias sudo='sudo '
alias l='ls -laFh '
```
The above whould allow running a command like `sudo l`

**Comment by waldauf** [comment]
I'm sorry, I was on holiday.

I use ZSH and I already have defined alias as you described. But it still doesn't work:
![image](https://github.com/zellij-org/zellij/assets/24603883/be39e3ab-8126-4d8a-ad78-15f7b0771ebd)

I also tried `sudo ll` which works as it should.

OS: MacOS
Terminal: Alacritty

**Comment by waldauf** [comment, marked as answer]
But after defining new alias for `zellij run -f -- ` it started to work:
![image](https://github.com/zellij-org/zellij/assets/24603883/08e881b5-ca02-44f0-87a2-017f884eb5c7)

Thanks for your help @Lockszmith-GH!
